src/flint/types/fmpq_mpoly.pyx

```diff
@@ -0,0 +1,982 @@
+from flint.flint_base.flint_base cimport flint_mpoly
+from flint.flint_base.flint_base cimport flint_mpoly_context
+
+from flint.utils.typecheck cimport typecheck
+from flint.utils.flint_exceptions import IncompatibleContextError
+
+from flint.types.fmpq cimport fmpq, any_as_fmpq
+from flint.types.fmpz_vec cimport fmpz_vec
+from flint.types.fmpq_vec cimport fmpq_vec
+
+from flint.types.fmpz cimport fmpz, any_as_fmpz
+from flint.types.fmpz_mpoly cimport fmpz_mpoly
+
+from flint.flintlib.flint cimport *
+from flint.flintlib.fmpq cimport fmpq_set, fmpq_one
+from flint.flintlib.fmpq_mpoly cimport *
+from flint.flintlib.fmpq_mpoly_factor cimport *
+
+from flint.flintlib.fmpz cimport fmpz_init_set
+from flint.flintlib.fmpz_mpoly cimport fmpz_mpoly_set
+
+from cpython.object cimport Py_EQ, Py_NE
+cimport cython
+cimport libc.stdlib
+
+cdef dict _fmpq_mpoly_ctx_cache = {}
+
+
+@cython.auto_pickle(False)
+cdef class fmpq_mpoly_ctx(flint_mpoly_context):
```

Comment:
Is this type picklable?

Reply:
No, almost none of these objects are because they contain a non-convertible member, the C context objects.
> Cython will generate a __reduce__() method to allow pickling an extension type if and only if each of its members are convertible to Python and it has no __cinit__ method.

That decorator can be removed though.

Reply:
Can we make the context objects picklable?